File: Assignment__14/Question3.c
//Write a program which accept number from user and check whether it contains 0 in it or not. 

#include<stdio.h>
#include<stdbool.h>

int Display(int iNo)
{
    int iDigit = 0;
    int FreqTwo = 0;

    if(iNo < 0)
    {
        iNo = -iNo;
    }

    while(iNo !=0)
    {
        iDigit = iNo % 10;

        if(iDigit == 2) 
        {                       
            FreqTwo = FreqTwo + 1;
        }                       
        iNo = iNo / 10;  
    }

    return FreqTwo;
}

int main()
{
    int iValue = 0;
    int iResult = 0;

    printf("Enter the number : \n");
    scanf("%d", &iValue);

    iResult = Display(iValue);

    printf("The number of twos are : %d\n", iResult);
    
    return 0;
}

File: Assignment__20/Question3.java
//Write a program to check whether a number is a perfect number or not.

import java.util.Scanner;

class Function
{
    int CheckPerfectNum(int iNo)
    {
        int iCnt = 0;
        int iSum = 0;
        for(iCnt = 1; iCnt <= iNo/2; iCnt++)
        {
            if(iNo % iCnt == 0)
            {
                iSum = iSum + iCnt;
            }
        }
        return iSum;
    }
}

class Question3
{
    public static void main(String A[])
    {
        Scanner sobj = new Scanner(System.in);
        
        System.out.println("Enter the number : ");
        int iValue = sobj.nextInt();

        int iResult = 0;

        Function fobj = new Function();
        iResult = fobj.CheckPerfectNum(iValue);

        if(iResult == iValue)
        {
            System.out.println("The number is perfect.");
        }
        else
        {
            System.out.println("The number is not perfect.");
        }
    }

}
